### PR TITLE
feat: add credential expiry validator

### PR DESCRIPTION
**Pull request by skynet2**

### CODE REVIEW

**Comment by mishasizov-SK** [comment]
@skynet2 please squash commits

File: pkg/doc/vc/validate_test.go [hunk 1]
```diff
@@ -136,10 +138,33 @@ func TestValidateCredential(t *testing.T) {
...
+					return "eyJhbGciOiJub25lIiwidHlwIjoiSldUIn0.eyJleHAiOjE1Nzc5MDY2MDQsImlhdCI6MTI2MjM3MzgwNCwiaXNzIjoiZGlkOmV4YW1wbGU6NzZlMTJlYzcxMmViYzZmMWMyMjFlYmZlYjFmIiwianRpIjoiaHR0cDovL2V4YW1wbGUuZWR1L2NyZWRlbnRpYWxzLzE4NzIiLCJuYmYiOjEyNjIzNzM4MDQsInN1YiI6ImRpZDpleGFtcGxlOmViZmViMWY3MTJlYmM2ZjFjMjc2ZTEyZWMyMSIsInZjIjp7IkBjb250ZXh0IjpbImh0dHBzOi8vd3d3LnczLm9yZy8yMDE4L2NyZWRlbnRpYWxzL3YxIiwiaHR0cHM6Ly93d3cudzMub3JnLzIwMTgvY3JlZGVudGlhbHMvZXhhbXBsZXMvdjEiLCJodHRwczovL3czaWQub3JnL3NlY3VyaXR5L2Jicy92MSJdLCJjcmVkZW50aWFsU3ViamVjdCI6eyJkZWdyZWUiOnsidHlwZSI6IkJhY2hlbG9yRGVncmVlIiwidW5pdmVyc2l0eSI6Ik1JVCJ9LCJpZCI6ImRpZDpleGFtcGxlOmViZmViMWY3MTJlYmM2ZjFjMjc2ZTEyZWMyMSIsIm5hbWUiOiJKYXlkZW4gRG9lIiwic3BvdXNlIjoiZGlkOmV4YW1wbGU6YzI3NmUxMmVjMjFlYmZlYjFmNzEyZWJjNmYxIn0sImV4cGlyYXRpb25EYXRlIjoiMjAyMC0wMS0wMVQxOToyMzoyNFoiLCJpZCI6Imh0dHA6Ly9leGFtcGxlLmVkdS9jcmVkZW50aWFscy8xODcyIiwiaXNzdWFuY2VEYXRlIjoiMjAxMC0wMS0wMVQxOToyMzoyNFoiLCJpc3N1ZXIiOnsiaWQiOiJkaWQ6ZXhhbXBsZTo3NmUxMmVjNzEyZWJjNmYxYzIyMWViZmViMWYiLCJuYW1lIjoiRXhhbXBsZSBVbml2ZXJzaXR5In0sInJlZmVyZW5jZU51bWJlciI6OC4zMjk0ODQ3ZSswNywidHlwZSI6WyJWZXJpZmlhYmxlQ3JlZGVudGlhbCIsIlVuaXZlcnNpdHlEZWdyZWVDcmVkZW50aWFsIl19fQ." // nolint
```
Comment (mishasizov-SK) [inline pkg/doc/vc/validate_test.go:145]: pls make it as go:embed testdata file

Comment (skynet2) [reply pkg/doc/vc/validate_test.go]: done @mishasizov-SK

File: pkg/doc/vc/validate.go [hunk 2]
```diff
@@ -26,5 +30,10 @@ func ValidateCredential(cred interface{}, formats []vcsverifiable.Format,
 		return nil, resterr.NewValidationError(resterr.InvalidValue, "credential", err)
 	}
 
+	if credential.Expired != nil && time.Now().UTC().After(credential.Expired.Time) {
+		return nil, resterr.NewValidationError(resterr.InvalidValue, "credential",
+			errors.New("credential expired"))
+	}
+
```
[the remark this answers is not in the document]
Comment (skynet2) [reply pkg/doc/vc/validate.go]: done @sudeshrshetty  https://github.com/trustbloc/vcs/pull/1080/files#diff-06589eafb6cce31687ef3ff9d52928826e3c779cd0597aedb8f723469a4d14d8R129

**Review by sudeshrshetty — APPROVED** [review]

**Review by mishasizov-SK — APPROVED** [review]

**Pull request merged**